src/cmd/flash.js

```diff
@@ -24,16 +27,42 @@ module.exports = class FlashCommand {
 
 		let result;
 		if (usb){
-			result = this.flashDfu({ binary, factory, force });
+			if (files.length > 0) {
+				// If both a device and a binary were provided, the binary will come in as the
+				// first element in the files array
+				binary = files[0];
+			} else {
+				// Otherwise, no device has been provided so unset the inaccurate value
+				device = undefined;
+			}
+
+			// Lookup the Device ID based on the provided Device Name
+			if (device && !isDeviceId(device)) {
+				const api = new ParticleApi(settings.apiUrl, { accessToken: settings.access_token }).api;
+
+				let deviceInfo;
+				try {
+					deviceInfo = await getDevice({ id: device, api, auth: settings.access_token });
```

Comment:
`getDevice()` from `device-util` doesn't really do anything special - i'd love it if we could just use the `ParticleApi` bits directly so eventually i can just remove `device-util` entirely.